monetio/models/_rrfs_sd_mm.py

```diff
@@ -0,0 +1,184 @@
+        cost (True) or not (False).
+
+    Returns
+    -------
+    xarray.DataSet
+        RRFS-SD model dataset in standard format for use in MELODIES-MONET
+
+    """
+
+    # Read in all variables and do all calculations.
+    dset = xr.open_mfdataset(fname, concat_dim="time", combine="nested", **kwargs)
+    
+    if 'smoke' in dset.data_vars and 'dust' in dset.data_vars:
+        dset['PM25'] = dset.smoke + dset.dust
+        dset['PM25'].attrs['long_name'] = 'Particulate Matter < 2.5 microns'
+        dset['PM25'].attrs['units'] = "ug/kg"
+    if 'smoke' in dset.data_vars and 'dust' in dset.data_vars and 'coarsepm' in dset.data_vars: 
+        dset['PM10'] = dset.smoke + dset.dust + dset.coarsepm
+        dset['PM10'].attrs['long_name'] = 'Particulate Matter < 10 microns'
+        dset['PM10'].attrs['units'] = "ug/kg"
+
+    # Standardize some variable names
+    dset = dset.rename(
+        {
+            "grid_yt": "y",
+            "grid_xt": "x",
+            "pfull": "z",
+            "phalf": "z_i",  # Interface pressure levels
+            "lon": "longitude",
+            "lat": "latitude",
+            "tmp": "temperature_k",  # standard temperature (kelvin)
+            "pressfc": "surfpres_pa",
+            "dpres": "dp_pa",  # Change names so standard surfpres_pa and dp_pa
+            "hgtsfc": "surfalt_m",
+            "delz": "dz_m",
+        }
+    )
+
+    # Calculate pressure. This has to go before sorting because ak and bk
+    # are not sorted as they are in attributes
+    dset["pres_pa_mid"] = _calc_pressure(dset,surf_only)
+
+    # Adjust pressure levels for all models such that the surface is first.
+    dset = dset.sortby("z", ascending=False)
+    dset = dset.sortby("z_i", ascending=False)
+
+    # Note this altitude calcs needs to always go after resorting.
+    # Altitude calculations are all optional, but for each model add values that are easy to calculate.
+    dset["alt_msl_m_full"] = _calc_hgt(dset)
+    dset["dz_m"] = dset["dz_m"] * -1.0  # Change to positive values.
+
+    # Set coordinates
+    dset = dset.reset_index(
+        ["x", "y", "z", "z_i"], drop=True
+    )  # For now drop z_i no variables use it.
+    dset["latitude"] = dset["latitude"].isel(time=0)
+    dset["longitude"] = dset["longitude"].isel(time=0)
+    dset = dset.reset_coords()
+    dset = dset.set_coords(["latitude", "longitude"])
+
+    # These sums and units are quite expensive and memory intensive,
+    # so add option to shrink dataset to just surface when needed
+    if surf_only:
+        dset = dset.isel(z=0).expand_dims("z", axis=1)
+
+    
+
+    # convert "ug/kg to ug/m3"
+    for i in dset.variables:
+        if "units" in dset[i].attrs:
+            if "ug/kg" in dset[i].attrs["units"]:
+                # ug/kg -> ug/m3 using dry air density
+                dset[i] = dset[i] * dset["pres_pa_mid"] / dset["temperature_k"] / 287.05535
+                dset[i].attrs["units"] = r"$\mu g m^{-3}$"
```

Comment:
If we really want a pretty version here could use unicode, e.g. `μg/m³` or `μg m⁻³`. But maybe better to use just ASCII `ug/m3` or `ug m-3`.

Reply:
Let’s stick with ascii. It may cause issues down the pipeline if we save the raw files out. It might not be recognized as CF convention otherwise

Reply:
Then I think `ug m-3` best for CF (though they would want `kg m-3` for canonical).